client/src/util/auth/Authentication.ts

```diff
@@ -26,14 +28,56 @@ export async function signOut() {
 
   if (auth.user) {
     const idToken = auth.user.id_token; // camelCase for variable name
-    localStorage.clear();
-    sessionStorage.clear();
-
-    await auth.removeUser();
-    await auth.signoutRedirect({
-      post_logout_redirect_uri: LOGOUT_URL.toString(),
-      id_token_hint: idToken,
-    });
+    const accessToken = auth.user.access_token; //console log to see if this works
+
+    try {
+      // Sign out silently
+      await auth.signoutSilent();
+
+      // Revoke tokens using the OAuth revoke API
+      await auth.revokeTokens();
+      const clientSecret = process.env.REACT_APP_CLIENT_SECRET; // TELL ARYAN
+
+      if (clientSecret) {
+        const revokeBody = {
+          token: accessToken,
+          token_type_hint: 'access_token',
+          client_id: process.env.REACT_APP_CLIENT_ID,
+          client_secret: clientSecret,
+        };
+
+        await fetch(`${process.env.REACT_APP_URL}/_oauth/logout`, {
```

Comment:
The signout happens via a GET request to this end point. There is no need for a request body. The required information goes via the headers. Please try [developer setup](https://github.com/INTO-CPS-Association/DTaaS/blob/feature/distributed-demo/docker/README.md) and then check `localhost/_oauth/logout`. Remember to change the the line-11 of `libms.dockerfile` to

```
COPY ./deploy/config/lib.docker .
```

You can then use the browser developer tools to check the headers going to the servers.